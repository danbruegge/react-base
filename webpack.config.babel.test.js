import webpackConfig from './webpack.config.babel';

jest.mock('path', () => ({
    resolve: (...args) => args[1],
    join: (...args) => args[1],
}));

test('Should match snapshot for development', () => {
    const env = { development: 'development' };
    const actual = webpackConfig(env);

    expect(actual).toMatchSnapshot();
});

test('Should match snapshot for production', () => {
    const env = { production: 'production' };
    const actual = webpackConfig(env);

    expect(actual).toMatchSnapshot();
});
