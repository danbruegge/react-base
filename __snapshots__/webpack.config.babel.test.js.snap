// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot for development 1`] = `
Object {
  "context": "src",
  "devServer": Object {
    "contentBase": "dist",
    "historyApiFallback": true,
    "host": "0.0.0.0",
    "port": 9000,
    "stats": "minimal",
  },
  "devtool": "cheap-module-eval-source-map",
  "entry": "./index.jsx",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "pre",
        "exclude": /\\(node_modules\\|dist\\)/,
        "loader": "eslint-loader",
        "options": Object {
          "failOnError": false,
          "fix": true,
          "rules": Object {
            "no-console": 0,
          },
        },
        "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
      },
      Object {
        "exclude": /\\(node_modules\\|dist\\)/,
        "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        "use": "babel-loader",
      },
    ],
  },
  "output": Object {
    "filename": "[name].js",
    "path": "dist",
    "publicPath": "/",
  },
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "IS_PRODUCTION": "false",
      },
    },
    ModuleConcatenationPlugin {
      "options": Object {},
    },
    Object {
      "apply": [Function],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": true,
        "inject": true,
        "minify": false,
        "showErrors": true,
        "template": "src/index.html",
        "title": "Webpack App",
        "xhtml": false,
      },
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;

exports[`Should match snapshot for production 1`] = `
Object {
  "context": "src",
  "devServer": Object {
    "contentBase": "dist",
    "historyApiFallback": true,
    "host": "0.0.0.0",
    "port": 9000,
    "stats": "minimal",
  },
  "devtool": "hidden-source-map",
  "entry": "./index.jsx",
  "module": Object {
    "rules": Array [
      Object {
        "enforce": "pre",
        "exclude": /\\(node_modules\\|dist\\)/,
        "loader": "eslint-loader",
        "options": Object {
          "failOnError": true,
          "fix": true,
          "rules": Object {
            "no-console": Array [
              2,
              Object {
                "allow": Array [
                  "warn",
                  "error",
                ],
              },
            ],
          },
        },
        "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
      },
      Object {
        "exclude": /\\(node_modules\\|dist\\)/,
        "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        "use": "babel-loader",
      },
    ],
  },
  "output": Object {
    "filename": "[name].js",
    "path": "dist",
    "publicPath": "/",
  },
  "plugins": Array [
    DefinePlugin {
      "definitions": Object {
        "IS_PRODUCTION": "true",
      },
    },
    ModuleConcatenationPlugin {
      "options": Object {},
    },
    Object {
      "apply": [Function],
    },
    HtmlWebpackPlugin {
      "options": Object {
        "cache": true,
        "chunks": "all",
        "compile": true,
        "excludeChunks": Array [],
        "favicon": false,
        "filename": "index.html",
        "hash": true,
        "inject": true,
        "minify": false,
        "showErrors": true,
        "template": "src/index.html",
        "title": "Webpack App",
        "xhtml": false,
      },
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".jsx",
    ],
  },
}
`;
